Improve propertiesExist assertion

bicpi opened the pull request:
From reading the code of the `propertiesExists()` assertion I thought it might not work as intended. 

As the foreach loop breaks at the first invalid property encountered it hardly makes sense to store the invalid properties in an array. First collecting all the properties and then displaying all of them is also what I'd expect as a user.

Moreover and for consistency reasons, I added the `isObjectOrClass()` check at the beginning –which is also present in the `propertyExists()` method.

(usually I like more fine grained unit testing including the exception messages etc. but I didn't want to break the current pattern; if desired I can add some more of course)

rquadling commented:
@bicpi Thank you for your PR. If you want to improve the unit tests, please do. Always happy to accept contributions with unit tests!

bicpi commented:
@rquadling Don't want to be too intrusive, just made sure that
* one or multiple properties can be asserted
* added the missing exception code
* switched to the "invalid" naming pattern which seemed to be used more often and lets room for "negative" assertions like `propertyNotExists()`

abacaphiliac commented:
👍
